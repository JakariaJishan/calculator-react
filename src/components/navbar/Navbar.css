header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 20px 0;
}

nav ul {
  display: flex;
  justify-content: center;
  align-items: center;
}

li {
  list-style: none;
  margin: 10px;
}

a {
  text-decoration: none;
  color: black;
}

a:hover {
  color: #f5913e;
}

.active {
  color: #f5913e;
  text-decoration: dotted underline;
}

.brand {
  font-size: 50px;
  font-weight: bold;
  color: #f5913e;
}
