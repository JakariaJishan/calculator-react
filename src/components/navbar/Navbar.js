import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import './Navbar.css';

function Navbar() {
  return (
    <>
      <header>
        <Link to="/" className="brand">Calculator</Link>
        <nav>
          <ul>
            <li>
              <NavLink to="/">Home</NavLink>
            </li>
            <li>
              <NavLink to="/calculator">Calculator</NavLink>
            </li>
            <li>
              <NavLink to="/quote">Quote</NavLink>
            </li>
          </ul>
        </nav>
      </header>
    </>
  );
}

export default Navbar;
