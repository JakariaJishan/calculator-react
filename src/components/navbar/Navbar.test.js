import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Navbar from './Navbar';


describe('Navbar testing', ()=>{
    it('snapshot matching', ()=>{
        const tree = renderer.create(<BrowserRouter><Navbar/></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot()
    })
    
    test('Document testing', ()=>{
        render(<BrowserRouter><Navbar/></BrowserRouter>)
        const navbarUi = screen.getAllByText('Calculator');
        expect(navbarUi).toBeTruthy()
    })
})
