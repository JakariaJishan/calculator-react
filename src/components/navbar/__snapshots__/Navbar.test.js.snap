// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar testing snapshot matching 1`] = `
<header>
  <a
    className="brand"
    href="/"
    onClick={[Function]}
  >
    Calculator
  </a>
  <nav>
    <ul>
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          className=""
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li>
        <a
          className=""
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
